models/issue_list.go

```diff
@@ -154,37 +154,11 @@ func (issues IssueList) loadMilestones(e Engine) error {
 	return nil
 }
 
-func (issues IssueList) getAssigneeIDs() []int64 {
-	var ids = make(map[int64]struct{}, len(issues))
-	for _, issue := range issues {
-		if _, ok := ids[issue.AssigneeID]; !ok {
-			ids[issue.AssigneeID] = struct{}{}
-		}
-	}
-	return keysInt64(ids)
-}
-
 func (issues IssueList) loadAssignees(e Engine) error {
-	assigneeIDs := issues.getAssigneeIDs()
-	if len(assigneeIDs) == 0 {
-		return nil
-	}
-
-	assigneeMaps := make(map[int64]*User, len(assigneeIDs))
-	err := e.
-		In("id", assigneeIDs).
-		Find(&assigneeMaps)
-	if err != nil {
-		return err
-	}
-
-	for _, issue := range issues {
-		if issue.AssigneeID <= 0 {
-			continue
-		}
-		var ok bool
-		if issue.Assignee, ok = assigneeMaps[issue.AssigneeID]; !ok {
-			issue.Assignee = NewGhostUser()
+	for in := range issues {
+		err := issues[in].loadAssignees(e)
```

Comment:
I think that it is not a good practice to load the assignees in separate requests.
Use something like that what has been used in `loadAttachments`.

Reply:
But `loadAttachments` does exactly that? (Although with a join)

Reply:
But `loadAttachments` uses only one request to load all attachments of all issues and your solution sends as many requests as issues are in the list. The `JOIN` concept is in that case more efficient.

Reply:
I see...

I'll change it.

Reply:
Done.